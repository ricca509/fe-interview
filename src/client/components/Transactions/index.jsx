import React from 'react';
import PropTypes from 'prop-types';

const Transactions = ({ transactions = [] }) => {
  return (
    <div className="transactions">
      {transactions.length} transactions
    </div>
  );
};

Transactions.propTypes = {
  transactions: PropTypes.array,
};

export default Transactions;