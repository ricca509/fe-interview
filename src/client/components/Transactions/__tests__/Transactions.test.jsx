import React from 'react';
import Transactions from '../';
import { shallow } from 'enzyme';

describe('Transactions component', ()=> {
  it('should render the count of trasactions', () => {
    const props = {
      transactions: [
        {
          amount: 12.34,
          date: '2017-01-13'
        },
        {
          amount: 14.34,
          date: '2017-02-13'
        },
        {
          amount: 15.54,
          date: '2017-03-13'
        },
        {
          amount: 11.34,
          date: '2017-04-13'
        },
        {
          amount: 18.99,
          date: '2017-05-13'
        }
      ],
    };

    const wrapper = shallow(<Transactions {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toEqual('5 transactions');
  });
});