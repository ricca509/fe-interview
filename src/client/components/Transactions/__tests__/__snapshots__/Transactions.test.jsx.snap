// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions component should render the count of trasactions 1`] = `
<div>
  5
   transactions
</div>
`;
